content/chainguard/chainguard-images/how-to-use/init-containers.md

```diff
@@ -0,0 +1,252 @@
+
+After creating this Dockerfile, build the image. Note that in both this command and the following one, you will need to change the `$REGISTRY` and `$NAMESPACE` variables to reflect your own container registry and namespace. If you plan to use Docker Hub as your container registry, you can omit the `$REGISTRY` portion of the command:
+
+```shell
+docker build -f Dockerfile.init -t $REGISTRY/$NAMESPACE/nginx-init .
+```
+
+Following that, push the init container image to your registry so it can be included in a Kubernetes pod:
+
+```shell
+docker push $REGISTRY/$NAMESPACE/nginx-init
+```
+
+Once you have built and pushed this init image to your registry it is now available to be run inside pipelines where you have your `nginx.conf` available for shared mounting and configuration in the init step.
+
+### Setting up a Kubernetes configuration
+
+Since you are building the entrypoint logic into the init container image itself, the Kubernetes deployment YAML for your init container would follow a similar structure to this:
+
+```yaml
+initContainers:
+  - name: run-entrypoint
+    image: $REGISTRY/$NAMESPACE/nginx-init
+    volumeMounts:
+      - name: nginx-config
+        mountPath: /etc/nginx
+```
+
+This instructs the init container to execute the startup scripts for the nginx configuration in the `docker-entrypoint.d` directory. In your main nginx configuration for the deployment, it would follow a structure similar to this:
+
+```yaml
+containers:
+  - name: nginx
+    image: cgr.dev/chainguard/nginx:latest
+    ports:
+      - containerPort: 80
+    volumeMounts:
+      - name: nginx-config
+        mountPath: /etc/nginx
+```
+
+Because your nginx container would mount the configurations from the shared volumeMount nginx-config in this case, you would now avoid having to run startup steps outside of the init. 
+
+The following command creates a deployment manifest named `init-deployment.yaml` that performs these steps using the init container image you created earlier:
+
+```shell
+cat > init-deployment.yaml <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+ name: nginx-chainguard
+spec:
+ replicas: 1
+ selector:
+   matchLabels:
+     app: nginx-chainguard
+ template:
+   metadata:
+     labels:
+       app: nginx-chainguard
+   spec:
+     volumes:
+       - name: nginx-config
+         emptyDir: {}
+
+     initContainers:
+       - name: run-entrypoint
+         image: $REGISTRY/$NAMESPACE/nginx-init
+         volumeMounts:
+           - name: nginx-config
+             mountPath: /etc/nginx
+
+     containers:
+       - name: nginx
+         image: cgr.dev/chainguard/nginx:latest
+         ports:
+           - containerPort: 80
+         volumeMounts:
+           - name: nginx-config
+             mountPath: /etc/nginx
+EOF
+```
+
```

Comment:
Something odd is happening when I paste this command, the $REGISTRY and $NAMESPACE variables are removed somehow and I end up with this:

```
initContainers:
       - name: run-entrypoint
         image: //nginx-init
         volumeMounts:
           - name: nginx-config
             mountPath: /etc/nginx
```

I know you are supposed to replace them, but the placeholder helps identify the place to edit. The issue seems to be with the `$`, not sure how that works in the EOF block, but since these aren't real variables it would be better to remove the `$` also to avoid confusion